src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -0,0 +1,91 @@
+package com.ably.kafka.connect.client;
+
+import com.ably.kafka.connect.config.ChannelSinkConnectorConfig;
+import com.ably.kafka.connect.config.ConfigValueEvaluator;
+import com.ably.kafka.connect.mapping.ChannelSinkMapping;
+import com.ably.kafka.connect.mapping.MessageSinkMapping;
+import io.ably.lib.http.HttpCore;
+import io.ably.lib.http.HttpUtils;
+import io.ably.lib.rest.AblyRest;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.HttpPaginatedResponse;
+import io.ably.lib.types.Message;
+import io.ably.lib.types.Param;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+/**
+ * Ably Batch client based on REST API
+ */
+public class DefaultAblyBatchClient extends DefaultAblyClient {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultAblyBatchClient.class);
+
+    AblyRest restClient;
+
+    public DefaultAblyBatchClient(ChannelSinkConnectorConfig connectorConfig, ChannelSinkMapping channelSinkMapping,
+                                  MessageSinkMapping messageSinkMapping, ConfigValueEvaluator configValueEvaluator) {
+        super(connectorConfig, channelSinkMapping, messageSinkMapping, configValueEvaluator);
+        this.restClient = null;
+    }
+
+    @Override
+    public void connect(SuspensionCallback suspensionCallback) throws ConnectException, AblyException {
+
+        this.restClient = new AblyRest(connectorConfig.getString(ChannelSinkConnectorConfig.CLIENT_KEY));
+
+    }
+
+    /**
+     * Function that uses the REST API client to send batches.
+     * @param records list of sink records.
+     * @throws ConnectException
+     */
+    public void publishBatch(List<SinkRecord> records) throws ConnectException, AblyException {
+
+        BatchSpec batch = null;
```

Comment:
So, I think what you really want to do here is create a mapping of `chanelName` -> `List<Messages>`, where as much as possible, we've preserved the order of messages in the topic partition for a given channel. This will mean that if a user has decided to disable parallelism, they see messages appear in Ably in the same order they posted them to Kafka.

You then construct a `BatchSpec` per outgoing channel name and submit them as a list. You'll always be sending sets of a single channel name, as the destination channel for each message in determined by the `ChannelSinkMapping` and we don't support sending the same message to multiple channels with the channel mapping config. (There are other broadcast use cases where the ability to send the same set of messages to many channels is useful, but not within the Kafka Connector)

@paddybyers correct me if I'm wrong

Reply:
Added logic to group by channel name.